### ISSUE TITLE
[BUG] Emojis are not displayed correctly

### ISSUE THREAD

**Issue by ssergio-ll** [CollaboraOnline/online#7015]
### Describe the Bug

Have version 23.05.2.2 of Collabora.

In the latest version of Collabora Online (Docker), emojis are not displayed correctly. This can occur in both spreadsheets and documents. They are displayed either completely in black, with incorrect colors, or in grayscale.

In spreadsheets, I have noticed that the ones displayed in black change color when you click on the cell.

### Steps to Reproduce

1.  Create a spreadsheet.
2. Insert any emoji.

### Expected Behavior

That emojis are displayed correctly.

### Actual Behavior

"The emojis are not displayed correctly."

### Screenshots

Screen 1: https://i.imgur.com/SEtzjTq.png
Screen 2: https://i.imgur.com/FuRNkPt.png

![image](https://github.com/CollaboraOnline/online/assets/7240617/3a7d6c1f-b515-4199-bcc5-48fa35284be1)

**Comment by timur-g** [comment]
Plesse explain how you insert emoji, can you attach here sampe ODT and why this would be different from #5172?

**Comment by nsilent22** [comment]
I came across this problem a few days ago. Here is the dialog for inserting special characters, with Noto Color Emoji font selected:
Previous version of Collabora:
COOLWSD: 22.05.5.4 git hash: d58a5e28 (E)
LOKit: Collabora Office 22.05.5.3 git hash: 53a9745
Ubuntu 20.04.4 LTS 2c9677ad

![good](https://github.com/CollaboraOnline/online/assets/14313222/ef1ddbfe-919b-48ae-ab00-1aeeda6f2a30)

Upgraded version of Collabora:
COOLWSD: 23.05.5.4 git hash: fc0de5b (E)
LOKit: Collabora Office 23.05.5.4 git hash: 1681200
Ubuntu 22.04.3 LTS bb7d6278

![not-so-good](https://github.com/CollaboraOnline/online/assets/14313222/60705b9e-cbf7-44d3-8204-9e02924bd5e9)

The icons have their red and blue pixel components swapped and all three values are inverterd (eg. 0x00 is where 0xff should be).
Here is the same picture with red and blue channels swapped + inversion:

![mangled](https://github.com/CollaboraOnline/online/assets/14313222/8d16f1d8-10af-41c7-9179-4a4998e62d0e)

Ona last thing: here are the editbox and three cells with frogs:
1. The frog is displayed correctly in editobox.
2. The frog is all black when it is put 'alone' in the first row.
3. The frog is grayscaled when a space preceeds it.
4. The frog is with wrong colors during edition. When edition ends it turns all-black or grayscale, depending on other characters in the cell (as in the point 2 or 3).

![frogs](https://github.com/CollaboraOnline/online/assets/14313222/ddc37692-df1f-480c-809a-3e1c569f4326)

**Comment by thebearon** [comment]
This was OK in 23.05.1-3, and was wrong in 23.05.2-2, can be tested with the respective Docker images (however, the bug isn't related to Docker, just that old images are readily available on the hub).

Since this must be due to a core change, the following range is relevant:
https://cgit.freedesktop.org/libreoffice/core/log/?qt=range&q=cc6b8db..22110b2

The following commit in the range deals with pixel order, and is the prime suspect:
https://cgit.freedesktop.org/libreoffice/core/commit/?id=ad65368380f49d8cfd111ea393243fa4d4888afb
```
author		Caolán McNamara <caolan.mcnamara@collabora.com>	2023-06-26 16:56:21 +0100
committer	Michael Meeks <michael.meeks@collabora.com>	2023-07-05 12:56:03 +0200

add an --enable-cairo-rgba to set the order of pixels for internal cairo
```

**Comment by thebearon** [comment]
While it's not related to this particular issue, let me note a few older emoji-related bugs, and what fixed those, perhaps it will be useful info for someone down the line, when investigating what can affect how emojis are showing up.

----

Back in 2020, in COOL 4.2, with core based on upstream 6.2 version, emojis were shown as placeholders, but 6.4 was good, it seems the following commits made a difference there:
https://cgit.freedesktop.org/libreoffice/core/commit/?id=bd216254af4bb695afbd1e957d02a2a162a08106
https://cgit.freedesktop.org/libreoffice/core/commit/?id=3681e8d8e6270912090b22445e67899bc2207e39
https://cgit.freedesktop.org/libreoffice/core/commit/?id=cb54bb89494218589227246f1923d8a24ab1676a
https://cgit.freedesktop.org/libreoffice/core/commit/?id=dcf7792da2aa2a1ef774a124f7b21f68fff0fd15
https://cgit.freedesktop.org/libreoffice/core/commit/?id=29411e7c08ec249116ee0211888e47c8b6f57707

----

After the following upstream commit, emojis stopped being shown.
https://cgit.freedesktop.org/libreoffice/core/commit/?id=e1b51099ac306118ee88bf3b7c94621df2564133
```
author		Stephan Bergmann <sbergman@redhat.com>	2022-01-05 08:07:11 +0100
committer	Stephan Bergmann <sbergman@redhat.com>	2022-01-05 09:17:02 +0100

Explicitly build external/freetype --without-png
```

Since the above change didn't seem necessary in any case we've encountered. the fix was to revert it, and carry that forward in CP branches:
https://cgit.freedesktop.org/libreoffice/core/commit/?id=085d5100a5ff3260bd4a58f15ecea66ad546fb2f
```
author		Andras Timar <andras.timar@collabora.com>	2022-12-07 20:18:42 +0100
committer	Andras Timar <andras.timar@collabora.com>	2022-12-08 07:26:47 +0000

[cp] Revert "Explicitly build external/freetype --without-png"
```

**Comment by caolanm** [comment]
I think I now see what this is happening and https://gerrit.libreoffice.org/c/core/+/167290 will fix it

**Comment by caolanm** [comment]
fixes merged to 24.04 and 23.05
